promql/ast.go

```diff
@@ -231,6 +231,9 @@ type VectorMatching struct {
 	// On contains the labels which define equality of a pair
 	// of elements from the vectors.
 	On model.LabelNames
+	// Ignoring excludes the given label names from matching,
+	// rather than only using them.
+	Ignoring bool
```

Comment:
I agree that booleans should be avoided if reasonable in APIs. In implementations I don't feel as strongly. Especially if they it goes through the same code path anyway and we now have to keep track of yet another invariant.

Maybe we can find a more expressive naming.